### PR TITLE
Gh-379: Add JSON info page to User Guide

### LINKED ISSUE #379: Add User Guide Page on JSON
Following the restructuring of the documentation under #378 a page on JSON to ensure that the end user / developer is aware of what JSON is and how the project uses this technology.

Topics to include:
- Basic overview of JSON syntax (external doc links where applicable)
- General usage of JSON in the Gaffer API
- Other config files that might use it

### PR DESCRIPTION
**Pull request by tb06904**
Adds a page on JSON and an intro to it to the documentation under the user guide.

# Related Issue

- Resolve #379

### CODE REVIEW

File: docs/user-guide/gaffer-basics/what-is-json.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# What is JSON?
+
+JavaScript Object Notation is a commonly used data format used for storing and
```
Comment (cn337131) [inline docs/user-guide/gaffer-basics/what-is-json.md]:
```suggestion
JavaScript Object Notation is a commonly used data format that can be used for storing and
```

File: docs/user-guide/gaffer-basics/what-is-json.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+JSON objects are defined by the contents of a pair of braces: `{}`, these then
+contain the various comma separated key value pairs that make up a block of data
+to be stored or transported. JSON has a number of supported data types and some
+notable exceptions.
+
+### Supported Data Types
+
+- **Strings** - A simple string of characters e.g. `"A String"`.
+- **Numbers** - Number representations, mostly compliant with ISO 6093 with a
+  few exceptions that we won't go into here.
```
Comment (GCHQDeveloper314) [inline docs/user-guide/gaffer-basics/what-is-json.md]: I'm not sure this is relevant in the context of JSON usage in Gaffer. Serialisation is done by Jackson Databind and so the ISO standard is not very relevant. We don't really want to encourage any unusual formats which might not be compatible.

Comment (tb06904) [reply docs/user-guide/gaffer-basics/what-is-json.md]: Agreed its probably not relevant to the end user to know about this,  I think just a passing comment to mentioning Jackson could be added and this tidied up

**Review by GCHQDeveloper314 — APPROVED** [review]

File: docs/user-guide/gaffer-basics/what-is-json.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    on supported JSON data types and features.
+
+### Unsupported Data Types
+
+- **Dates** - There is no inbuilt date standard for JSON, most people use a
+  `String` to represent a date value.
+- **Undefined** - Undefined is a special state where something does not exist,
+  this is different to in concept to null.
+
+## How is JSON Used in Gaffer?
```
Comment (rj77259) [inline docs/user-guide/gaffer-basics/what-is-json.md]: Should this be a lower case u for Used?

File: docs/user-guide/gaffer-basics/what-is-json.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+JSON in Gaffer is largely used as an interchange format for data and can be found
+mainly in the API but also in many configuration files. The following sections
+give a quick introduction as to where and how JSON is used in Gaffer.
+
+### JSON in the Gaffer API
+
+Gaffer provides a number of communication methods so that you can
+programmatically interact with a graph, as of Version 2.0 this is provided in 3
+ways: A HTTP rest API, a Python API and a Native Java interface. Of these 3 both the
```
Comment (rj77259) [inline docs/user-guide/gaffer-basics/what-is-json.md]: Should rest be capitalized?

**Review by rj77259 — DISMISSED** [review]

**Pull request merged**